### PR TITLE
Add flaws check / fix to cli

### PR DESCRIPTION
**Pull request by fiji-flo**
- add flaws check to validate
- add flaws command

### CODE REVIEW

**Comment by peterbe** [comment]
What is the context here? When would you use this? Does it solve an issue we have?

**Comment by fiji-flo** [comment]
The idea was to enable editors to fix flaws from the cli and add it to the validation.


Mostly parity with the UI though.

**Review by peterbe — APPROVED** [review]
Works! Clearly...
Screen Shot 2020-11-03 at 2 01 47 PM

You'd get the same effect/result if you visit http://localhost:3000/en-US/docs/Web/CSS/background-image#_flaws and press the "Fix fixable flaws (3)" button. 

The only thing I don't like is that `yarn tool flaws [slug]` feels like it would *show* the flaws or something. So it doesn't feel super intuitively named. Perhaps `yarn tool fix-flaws [slug]` would make more sense. 

But I welcome the functionality. Perhaps it ought to be documented too. Nit?

The other thing I'm not entirely sure about is if the `slug` is the best entry for this. I think I'd rather be able to do:
```
▶ yarn tool flaws http://localhost:3000/en-US/docs/Web/CSS/background-image
```
or
```
▶ yarn tool flaws /en-US/docs/Web/CSS/background-image
```

But perhaps that's just me. And perhaps it would be somewhat trivial to hack around like this:

```javascript
      const { slug, locale } = args;
      const { yes } = options;
      let url;
      if (slug.includes('://') || slug.startsWith('/')) {
          if (slug.includes('://')) {
              url = new URL(slug).pathname
          } else {
            url=slug;
          }
      } else {
           url=buildURL(locale, slug)
      }
      const document = Document.findByURL(url);
```

**Comment by fiji-flo** [comment]
I'd like to add some more capabilities to that command in the future. For now I think `flaws` is an okay name as it also just displays flaws and fixing is (kind of) optional. I get you suggestion to make it no only accept slugs but this should then change for all commands.

I filed #1642 to follow up on this.

**Comment by peterbe** [comment]
Ok. Fair enough. I think this'll come in handy but only time will tell where our actual users gravitate and we won't know what for months. 
You still have my approval. Now we have to wait for @escattone

File: tool/cli.js
Comment (escattone) [inline tool/cli.js:226]: This feels redundant and a bit confusing to me (e.g., are issues different than flaws?), since before this message is shown, the user will see `Found X flaws.`. For example:`
```sh
rjohnson-44864:deployer rjohnson$(maws_profile)$ yarn tool validate Web
yarn run v1.22.5
$ node tool/cli.js validate Web
Found 4 flaws.
✗ Found some issues
✨  Done in 1.28s.
```
Could we simply skip this message when `okay` is falsy?

File: tool/cli.js
Comment (escattone) [inline tool/cli.js:207]: I think we need to show a friendly error message when the document can't be found.

File: tool/cli.js
Comment (escattone) [inline tool/cli.js:256]: Likewise here, a friendly error message when the document can't be found.

File: tool/cli.js
Comment (escattone) [inline tool/cli.js:272]: This is broken when `-y, --yes` is used on the command line:
```sh
rjohnson-44864:deployer rjohnson$(maws_profile)$ yarn tool flaws Web --yes
yarn run v1.22.5
$ node tool/cli.js flaws Web --yes
Would fix (link1) broken_link /en-US/docs/Web/Apps to /en-US/docs/Web/Progressive_web_apps
Would fix (link4) broken_link /en-US/docs/SVG to /en-US/docs/Web/SVG
Would modify "/Users/rjohnson/repos/content/files/en-us/web/index.html" from fixable flaws.
✨  Done in 1.30s.
```

Maybe something like:
```suggestion
      const run = yes
        ? true
        : await prompts({
            type: "confirm",
            message: `Proceed fixing ${flaws} flaws?`,
            name: "run",
            initial: true,
          }).run;
```

**Review by escattone — CHANGES_REQUESTED** [review]
@fiji-flo I really like this! I think it needs to show a more user-friendly error message though when the user provides a slug that doesn't exist. Currently it shows `Cannot destructure property 'metadata' of 'document' as it is undefined.`:
```sh
rjohnson-44864:deployer rjohnson$(maws_profile)$ yarn tool validate Web/ZZZ
yarn run v1.22.5
$ node tool/cli.js validate Web/ZZZ
Cannot destructure property 'metadata' of 'document' as it is undefined.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
rjohnson-44864:deployer rjohnson$(maws_profile)$ yarn tool flaws Web/ZZZ
yarn run v1.22.5
$ node tool/cli.js flaws Web/ZZZ
Cannot destructure property 'metadata' of 'document' as it is undefined.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by fiji-flo** [comment]
@escattone thanks for the feedback all should be addressed

File: tool/cli.js
Comment (escattone) [inline tool/cli.js:209]: Typo:
```suggestion
        throw new Error(`Slug ${slug} does not exist for ${locale}`);
```

File: tool/cli.js
Comment (escattone) [inline tool/cli.js:259]:
```suggestion
        throw new Error(`Slug ${slug} does not exist for ${locale}`);
```

**Review by escattone — APPROVED** [review]
👍 Two tiny typos to fix before you merge. Thanks @fiji-flo!

**Comment by escattone** [comment]
@fiji-flo If you don't mind, I'll just commit the tiny typo fixes and merge this. Thanks!

**Pull request merged**
